features/FxA-45-device-registration-api/README.md

```diff
@@ -0,0 +1,360 @@
+  },
+  {
+    "id": "01ab",
+    "name": "My Desktop",
+    "type": "desktop",
+    "status": "disconnected",
+    "callback": "https://updates.push.services.mozilla.com/update/d4c5b1e3f5791ef83896c27519979b93a45e6d0da34c75"
+  }
+]
+```
+
+## Details
+
+### What is a "recovering device"?
+
+A recovering device is one that
+has been registered before but has lost its device id.
+We may rely on the device `name` field in this case.
+The recovering device will login as a new device, submitting its name.
+If the name matches an existing device,
+we will consider them to be the same,
+otherwise a new device will be created.
+
+Responding to "We may rely on the device `name` field in this case",
+Ryan K commented:
+
+> I don't think I like this,
+> it seems too easy to have device names that accidentally match,
+> particularly if the device fills in a default based on its hardware platform etc.
+> If we're going to do recovering devices,
+> I think we should let the device provide some sort of fingerprint
+> that it's extremely confident will be unique.
+
+### How are shared devices handled?
+
+A shared device may or may not use the same device id for each user,
+there are tradeoffs for both.
+Either way, it should be difficult for the service to detect
+whether a device is shared and by whom.
+
+### Are shared devices bound to the account or the device?
+
+For example, Alice and Bob share a device.
+Alice registers the device with the name "Alice's Tablet".
+Later, Bob logs in and changes the name to "Bob's Tablet".
+The server still has her device name as "Alice's Tablet"
+but the device may have saved it locally as "Bob's Tablet".
+Should Alice's record be updated?
+When Alice logs in again, which name should she see?
+
+It can work either way, but we should decide on one or the other:
+
+  * Shared device with shared info;
+  * Or shared device with individual info?
+
+Danny's preference was for individual info.
+
+Ryan K commented:
+
+> It's not really clear how this should behave from device's individual perspective,
+> let alone when you put the server in the mix.
+> For e.g. a desktop firefox instance,
+> ISTM that "Bob logs in" implies that Alice gets logged out,
+> which could clear the locally-stored device name
+> and other FxA-related customizations.
+> For a dedicated Firefox device maybe it's not so simple.
+
+### How are unrecognised devices handled?
+
+If a login request specifies an unknown device id,
+the login may still succeed
+but the device field in the response will be `null`.
+This may indicate that the device was explicitly deleted
+from the device list
+by the user.
+Posting to `/v1/account/device` can attach a new device
+to the current session.
+
+### How big should the device id be?
+
+It must be:
+
+  1. Big enough to make collisions unlikely per account;
+  2. And small enough not to identify individual accounts.
```

Comment:
As per  [Ryan's comment](https://github.com/mozilla/fxa/pull/69/files#r43348663), this requirement no longer exists. Have deleted this section.